Add `dynamo_timed` to a few more spots in AOTAutograd

bdhirsh opened the pull request:
This should help us see a bit more clearly how expensive our `make_fx()` call and our metadata collection phase are in AOTAutograd

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #141489

bdhirsh committed:
include run_functionalized_fw_and_collect_metadata in compile events

[ghstack-poisoned]

bdhirsh committed:
Update on "include run_functionalized_fw_and_collect_metadata in compile events"




[ghstack-poisoned]

zou3519 commented on torch/_functorch/_aot_autograd/dispatch_and_compile_graph.py:
```diff
@@ -51,7 +51,7 @@ def _create_graph(f, args, *, aot_config: AOTConfig) -> torch.fx.GraphModule:
         # Allow token discovery for joint fn tracing as tokens can be used in backward.
         _allow_token_discovery=True,
-    ):
+    ), dynamo_timed("_create_graph", log_pt2_compile_event=True):
```
do we want a more detailed name for this or is `_create_graph` something generic?

bdhirsh replied:
I figured there was some convention for naming these after the function they're called in (@jamesjwu are there any conventions we should be following for the naming?). Otherwise, totally agreed - I can name it something clearer like "aotdispatcher_makefx")

jamesjwu replied:
Can we call this something like `dispatch_and_compile_graph._create_graph` to be more specific?

jamesjwu commented:
Do we need `log_pt2_compile_event=True` here? What that does is log it to Scuba, but I feel like just having it tlparse is already sufficient for these smaller events. Is there a reason you care about aggregating here? Keeping it off by default would still have them show up in the chromium trace, but won't impact table retention.

jamesjwu commented:
Also, could you run tlparse on a benchmark or two and link what the tlparse/chromium event looks like? Just to make sure it looks as you expect and there aren't any other big gaps to track

bdhirsh commented:
I'm having some environment issues so I can't easily run a huggingface benchmark right now, but with this simple benchmark:
```
@torch.compile
def fn(x):
    for _ in range(1000):
        x = x.sin()
    return x

x = torch.ones(16, device="cuda", requires_grad=True)
out = fn(x)
```

This is the profile I get: https://interncache-all.fbcdn.net/manifold/perfetto-artifacts/tree/ui/index.html?url=https%3A%2F%2Finterncache-all.fbcdn.net%2Fmanifold%2Ftlparse_reports%2Ftree%2Flogs%2F.tmptkyNA4%2Fdedicated_log_torch_trace_pw4m56vb.log%2Fchromium_events.json#!/viewer?url=https%3A%2F%2Finterncache-all.fbcdn.net%2Fmanifold%2Ftlparse_reports%2Ftree%2Flogs%2F.tmptkyNA4%2Fdedicated_log_torch_trace_pw4m56vb.log%2Fchromium_events.json&local_cache_key

you can see that we have roughly:

~1s in `run_functionalized_fw_and_collect_metadata`
~5s in `make_fx`
~1s in `partition_fn`

bdhirsh committed:
Update on "Add `dynamo_timed` to a few more spots in AOTAutograd"

This should help us see a bit more clearly how expensive our `make_fx()` call and our metadata collection phase are in AOTAutograd




[ghstack-poisoned]

bdhirsh commented:
@pytorchbot merge

bdhirsh committed:
Update on "Add `dynamo_timed` to a few more spots in AOTAutograd"

This should help us see a bit more clearly how expensive our `make_fx()` call and our metadata collection phase are in AOTAutograd




[ghstack-poisoned]
